include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -38,10 +38,92 @@
 namespace seqan3
 {
 
+/*!\brief Computes a pairwise alignment over a sequence pair or a range of sequence pairs.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam sequence_t         The type of sequence pairs. Either a type modelling seqan3::tuple_like_concept
+ *                            with a tuple size of exactly two or std::ranges::InputRange whose value type models the
+ *                            before mentioned tuple concept.
+ * \tparam alignment_config_t The type of the alignment configuration; must be a seqan3::configuration.
+ * \param[in] seq             A tuple with two sequences or an input range over such tuples.
+ * \param[in] config          A configuration object containing \ref alignment_configuration
+ *                            "alignment configurations".
+ * \return Depending on the configuration, the return type of align_pairwise is either `void` if the alignment was
+ * configured to call a delegate on each hit or a seqan3::alignment_range over the alignment results.
+ *
+ * \details
+ *
+ * This function computes the pairwise alignment of a sequence pair or a range of sequence pairs.
+ * The type of the elements stored in the tuples must model std::ranges::ForwardRange.
+ * The value category of these elements is either an lvalue or if the passed tuple or range is an lvalue, xvalues are
+ * also allowed as tuple elements.
+ *
+ * For each alignment one or more alignment results can be computed depending on the configuration.
+ * Before computing the alignment the respective alignment algorithm is configured depending on the given alignment
+ * configuration. If the configuration enables a two-way execution (the default case), a seqan3::alignment_range is
+ * returned. The alignment is first computed when calling begin on the returned range or the respective iterator is
+ * incremented.
```

Comment:
I would maybe say that "alignments are computed on-demand  when iterating over the results", maybe that's easier to understand?